common/src/main/java/com/microsoft/identity/common/AndroidPlatformComponents.java

```diff
@@ -59,29 +67,82 @@
 /**
  * Android implementations of platform-dependent components in Common.
  */
-public class AndroidCommonComponents implements ICommonComponents {
-    private static String TAG = AndroidCommonComponents.class.getSimpleName();
+public class AndroidPlatformComponents implements IPlatformComponents {
+    private static final String TAG = AndroidPlatformComponents.class.getSimpleName();
 
     /**
      * SharedPref filename for Clock Skew storage.
      */
     private static final String SKEW_PREFERENCES_FILENAME =
             "com.microsoft.identity.client.clock_correction";
 
+    @NonNull
     protected final Context mContext;
+
+    @Nullable
+    protected final Activity mActivity;
+
+    @Nullable
+    protected final Fragment mFragment;
+
     private IClockSkewManager mClockSkewManager;
     private IDevicePopManager mDefaultDevicePoPManager;
 
-    public AndroidCommonComponents(@NonNull final Context context) {
-        mContext = context;
-        Device.setDeviceMetadata(new AndroidDeviceMetadata());
+    /**
+     * True if all of the platform-dependent static classes have been initialized.
+     */
+    private static boolean sInitialized = false;
+
+    /**
+     * Initializes platform-dependent static classes.
+     */
+    public static synchronized void initializeStaticClasses() {
+        if (!sInitialized) {
+            Device.setDeviceMetadata(new AndroidDeviceMetadata());
+            Logger.setAndroidLogger();
```

Comment:
It looks like these are not dependent at all on the members of the class.  Do we want to do this initialization in a static context instead of a checked variable at construction time?  If not, it's worth it to say why.

Reply:
Eventually - yes - at the entry point of broker.
Right now we're still working our way there, so I put it here. Will add todo to call this out.